pkgs/build-support/rust/default.nix

```diff
@@ -41,18 +41,25 @@ let
       cargoDepsCopy="$sourceRoot/${cargoVendorDir}"
     '';
 
+  nixConfig = stdenv.hostPlatform.config;
+  rustConfig = {
+    "x86_64-pc-mingw32" = "x86_64-pc-windows-gnu";
+  }."${nixConfig}" or nixConfig;
+
   ccForBuild="${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}cc";
   cxxForBuild="${buildPackages.stdenv.cc}/bin/${buildPackages.stdenv.cc.targetPrefix}c++";
   ccForHost="${stdenv.cc}/bin/${stdenv.cc.targetPrefix}cc";
   cxxForHost="${stdenv.cc}/bin/${stdenv.cc.targetPrefix}c++";
-  releaseDir = "target/${stdenv.hostPlatform.config}/${buildType}";
-in stdenv.mkDerivation (args // {
+  releaseDir = "target/${rustConfig}/${buildType}";
+in
+
+stdenv.mkDerivation (args // {
   inherit cargoDeps;
 
   patchRegistryDeps = ./patch-registry-deps;
 
-  nativeBuildInputs = [ cargo rustc git cacert ] ++ nativeBuildInputs;
-  inherit buildInputs;
+  nativeBuildInputs = nativeBuildInputs ++ [ cacert git cargo rustc ];
+  buildInputs = buildInputs ++ stdenv.lib.optional (nixConfig == "x86_64-pc-mingw32") windows.pthreads;
```

Comment:
isWindows or isMinGW?

Reply:
Changed to `stdenv.hostPlatform.isMinGW`, thanks! `pthreads` is unlikely to be a dependency in MSVC builds.